pymeasure/instruments/thermotron/thermotron3800.py

```diff
@@ -0,0 +1,129 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+class Thermotron3800(Instrument):
+    """ Represents the Thermotron 3800 Oven.
+    For now, this driver only supports using Control Channel 1.
+    """
+
+    def __init__(self, resourceName, **kwargs):
+        super().__init__(
+            resourceName,
+            "Thermotron 3800",
+            **kwargs
+        )
+
+    id = Instrument.measurement(
+        "IDEN?", """ Reads the instrument identification 
+        
+        :return: String
+        """
+    )
+
+    temperature = Instrument.measurement(
+        "PVAR1?", """ Reads the current temperature of the oven 
+        via built in thermocouple
+        
+        :return: float
+        """
+    )
+
+    mode = Instrument.measurement(
+        "MODE??", """ Gets the operating mode of the oven.
+        
+        :return: Tuple(String, int)
+        """,
+        get_process=lambda mode: Thermotron3800.__translate_mode(mode)
+    )
+    
+    setpoint = Instrument.control(
+        "SETP1?", "SETP1,%g",
+        """ A floating point property that controls the setpoint 
+        of the oven in Celsius. This property can be set. 
+        
+        Insert wait time after this command. This wait time should be >1000ms for consistent results.
+        Failing to wait for an adequate time may cause errors in subsequent oven commands.
```

Comment:
Thank you! I really like this solution. I have implemented it into my driver, and updated the comments for the other commands. 

I am a bit unsure what to do about the ```initialize_oven``` command, though. The manufacturer recommends a 3 second wait time after sending this commands. This is a long time to block the user, while they could be calling other instrument initialization functions and take care of the wait themselves. If there are multiple ovens to initialize, it would take 6 seconds, as opposed to just 3 seconds if the user has control of the wait (send both init commands and then insert the wait after). Should I incorporate the wait time into the init function, or allow the user to have control? Or would it be better to incorporate it into the function, but have an optional argument to disable the wait?

Reply:
Hi msmttchr, I added an optional wait argument to the initialize_oven function. The wait time from the initialize_oven function is 2 seconds, for a total of 3 when calling the initalize_oven command (2 seconds from the function + 1 second built into the write function).